docs/standalone.md

```diff
@@ -0,0 +1,354 @@
+
+```c
+valkeyContext *valkeyConnect(const char *host, int port);
+valkeyContext *valkeyConnectUnix(const char *path);
+
+// There is also a convenience struct to specify various options.
+valkeyContext *valkeyConnectWithOptions(valkeyOptions *opt);
+```
+
+When connecting to a server, libvalkey will return `NULL` in the event that we can't allocate the context, and set the `err` member if we can connect but there are issues. So when connecting it's simple to handle error states.
+
+```c
+valkeyContext *ctx = valkeyConnect("localhost", 6379);
+if (ctx == NULL || ctx->err) {
+    fprintf(stderr, "Error connecting: %s\n", ctx ? ctx->errstr : "OOM");
+}
+```
+
+### Connection options
+
+There are a variety of options you can specify when connecting to the server, which are delivered via the `valkeyOptions` helper struct. This includes information to connect to the server as well as other flags.
+
+```c
+valkeyOptions opt = {0};
+
+// You can set primary connection info
+if (tcp)  {
+    VALKEY_OPTIONS_SET_TCP(&opt, "localhost", 6379);
+} else {
+    VALKEY_OPTIONS_SET_UNIX(&opt, "/tmp/redis.sock");
+}
+
+// You may attach any arbitrary data to the context
+VALKEY_OPTIONS_SET_PRIVDATA(&opt, my_data);
+```
+
+There are also several flags you can specify when using the `valkeyOptions` helper struct.
+
+| Flag | Description  |
+| --- | --- |
+| VALKEY\_OPT\_NONBLOCK | Tells libvalkey to make a non-blocking connection. |
+| VALKEY\_OPT\_REUSEADDR | Tells libvalkey to set the [SO_REUSEADDR](https://man7.org/linux/man-pages/man7/socket.7.html) socket option |
+| VALKEY\_OPT\_PREFER\_IPV4<br>VALKEY\_OPT\_PREFER_IPV6<br>VALKEY\_OPT\_PREFER\_IP\_UNSPEC | Informs libvalkey to either prefer IPv4 or IPv6 when invoking [getaddrinfo](https://man7.org/linux/man-pages/man3/gai_strerror.3.html).  `VALKEY_OPT_PREFER_IP_UNSPEC` will cause libvalkey to specify `AF_UNSPEC` in the getaddrinfo call, which means both IPv4 and IPv6 addresses will be searched simultaneously.<br>Libvalkey prefers IPv4 by default. |
+| VALKEY\_OPT\_NO\_PUSH\_AUTOFREE | Tells libvalkey to not install the default RESP3 PUSH handler (which just intercepts and frees the replies).  This is useful in situations where you want to process these messages in-band. |
+| VALKEY\_OPT\_NOAUTOFREEREPLIES | **ASYNC**: tells libvalkey not to automatically invoke `freeReplyObject` after executing the reply callback. |
+| VALKEY\_OPT\_NOAUTOFREE | **ASYNC**: Tells libvalkey not to automatically free the `valkeyAsyncContext` on connection/communication failure, but only if the user makes an explicit call to `valkeyAsyncDisconnect` or `valkeyAsyncFree` |
+
+### Executing commands
+
+The primary command interface is a `printf`-like function that takes a format string along with a variable number of arguments. This will construct a resp command and deliver it to the server.
```

Comment:
```suggestion
The primary command interface is a `printf`-like function that takes a format string along with a variable number of arguments. This will construct a `RESP` command and deliver it to the server.
```